CRM/Contact/Form/Task/SaveSearch.php

```diff
@@ -232,6 +236,20 @@ public function postProcess() {
 
     CRM_Contact_BAO_Group::create($params);
 
+    // Update mapping with the name and description of the group.
+    if ($mappingId) {
+      $mappingParams = array(
+        'id' => $mappingId,
+        'name' => CRM_Core_DAO::getFieldValue('CRM_Contact_DAO_Group', $group->id, 'name', 'id'),
```

Comment:
Steps I used to replicate : 
- On Search Builder searched contacts with criteria ```gender = Male``
- Chosen 'Create Smart Group' Task on result list
- Redirected Smart Group form, filled up description and name 
- Upon submit, I got a fatal error (backtrace is recorded in configLog)